app/views/SelectedUsersView/index.tsx

```diff
@@ -0,0 +1,179 @@
+import { Q } from '@nozbe/watermelondb';
+import orderBy from 'lodash/orderBy';
+import React, { useCallback, useEffect, useLayoutEffect, useState } from 'react';
+import { FlatList } from 'react-native';
+import { shallowEqual, useDispatch } from 'react-redux';
+import { Subscription } from 'rxjs';
+import { RouteProp, useNavigation, useRoute } from '@react-navigation/native';
+import { StackNavigationProp } from '@react-navigation/stack';
+
+import { addUser, removeUser, reset } from '../../actions/selectedUsers';
+import * as HeaderButton from '../../containers/HeaderButton';
+import * as List from '../../containers/List';
+import { sendLoadingEvent } from '../../containers/Loading';
+import SafeAreaView from '../../containers/SafeAreaView';
+import StatusBar from '../../containers/StatusBar';
+import { ISearch, ISearchLocal } from '../../definitions';
+import I18n from '../../i18n';
+import database from '../../lib/database';
+import UserItem from './UserItem';
+import { ISelectedUser } from '../../reducers/selectedUsers';
+import { getUserSelector } from '../../selectors/login';
+import { ChatsStackParamList } from '../../stacks/types';
+import { useTheme } from '../../theme';
+import { showErrorAlert } from '../../lib/methods/helpers/info';
+import log, { events, logEvent } from '../../lib/methods/helpers/log';
+import { search as searchMethod } from '../../lib/methods';
+import { isGroupChat as isGroupChatMethod } from '../../lib/methods/helpers';
+import { useAppSelector } from '../../lib/hooks';
+import Header from './Header';
+
+type TRoute = RouteProp<ChatsStackParamList, 'SelectedUsersView'>;
+type TNavigation = StackNavigationProp<ChatsStackParamList, 'SelectedUsersView'>;
+
+type TSearchItem = ISearch | ISearchLocal;
+
+const SelectedUsersView = () => {
+	const [chats, setChats] = useState<ISelectedUser[]>([]);
+	const [search, setSearch] = useState<TSearchItem[]>([]);
+
+	const { maxUsers, showButton, title, buttonText, nextAction } = useRoute<TRoute>().params;
+	const navigation = useNavigation<TNavigation>();
+
+	const { colors } = useTheme();
+	const dispatch = useDispatch();
+
+	const { users, loading, useRealName, user } = useAppSelector(
+		state => ({
+			users: state.selectedUsers.users,
+			loading: state.selectedUsers.loading,
+			useRealName: state.settings.UI_Use_Real_Name as boolean,
+			user: getUserSelector(state)
+		}),
+		shallowEqual
+	);
+
+	sendLoadingEvent({ visible: loading });
+
+	const isChecked = useCallback((username: string) => users.findIndex(el => el.name === username) !== -1, [users]);
```

Comment:
This is being recreated on every render, because users is a non-primitive value.
Why do you need the useCallback?
You're wasting time trying to improve the performance of something without knowing if it's needed.